magnet_coils.py

```diff
@@ -0,0 +1,247 @@
+            cubit.cmd(
+                'rotate Surface ' + str(surf_id) + ' angle ' + str(rot_ang_norm)
+                + ' about origin 0 0 0 direction ' + ' '.join(str(i) for i in
+                rot_axis) + ' include_merged'
+            )
+
+            # If cross-section shape is rectangular, orient cross-section in
+            # normal plane such that its vector pointing in the direction of its
+            # thickness faces the origin
+            if shape == 'rectangle':
+                # Extract initial path point
+                pos = cubit.vertex(path[0]).coordinates()
+
+                # Cross-section inititally populated with thickness vector
+                # oriented along x axis
+                t_vec = np.array([1, 0, 0])
+                # Compute part of thickness vector parallel to rotation axis
+                t_vec_par = (np.inner(t_vec, rot_axis)/np.inner(rot_axis,
+                    rot_axis))*rot_axis
+                # Compute part of thickness vector orthogonal to rotation axis
+                t_vec_perp = t_vec - t_vec_par
+                # Compute magnitude of orthogonal part of thickness vector
+                t_vec_perp_mag = np.linalg.norm(t_vec_perp)
+
+                # Compute vector othogonal to both rotation axis and orthogonal
+                # part of thickness vector
+                orth = np.cross(rot_axis, t_vec_perp)
+                # Compute magnitude of orthgonal vector
+                orth_mag = np.linalg.norm(orth)
+
+                # Convert angle from degress to radians
+                rot_ang_norm = np.deg2rad(rot_ang_norm)
+                
+                # Determine part of rotated vector parallel to original
+                rot_par = np.cos(rot_ang_norm)/t_vec_perp_mag
+                # Determine part of rotated vector orthogonal to original
+                rot_perp = np.sin(rot_ang_norm)/orth_mag
+
+                # Compute orthogonal part of thickness vector after rotation
+                t_vec_perp_rot = t_vec_perp_mag*(rot_par*t_vec_perp
+                    + rot_perp*orth)
+                # Compute thickness vector after rotation
+                t_vec_rot = t_vec_perp_rot + t_vec_par
+                # Compute magnitude of thickness vector
+                t_vec_rot_mag = np.linalg.norm(t_vec_rot)
+
+                # Project position vector onto cross-section
+                pos_proj = pos - (np.inner(pos, norm)/np.inner(pos, pos))*norm
+                # Compute magnitude of projection of position vector
+                pos_proj_mag = np.linalg.norm(pos_proj)
+
+                # Compute angle by which to rotate cross-section such that it
+                # faces the origin
+                dot_prod = np.inner(pos_proj, t_vec_rot)
+                rot_ang_orig = np.arccos(dot_prod/t_vec_rot_mag/pos_proj_mag)
+
+                # Convert angle from radians to degrees
+                rot_ang_orig = np.rad2deg(rot_ang_orig)
```

Comment:
The `cubit_export` function and `magnet_coils` script provide different functionality, so my thought here was to initialize Cubit separately in case one would like to use one and not the other.